schema/link.go

```diff
@@ -124,6 +124,38 @@ func (r ResourceLink) GetEndpoints() []Endpoint {
 	return eps
 }
 
+// GetSecureEndpoints returns secure endpoints in order of priority.
+func (r ResourceLink) GetSecureEndpoints() []Endpoint {
+	endpoints := make([]Endpoint, 0, 4)
+	for _, ep := range r.GetEndpoints() {
+		addr, err := ep.GetAddr()
+		if err != nil {
+			continue
+		}
+		switch addr.GetScheme() {
+		case string(TCPSecureScheme), string(UDPSecureScheme):
+			endpoints = append(endpoints, ep)
+		}
+	}
+	return endpoints
+}
+
+// GetUnsecureEndpoints returns unsecure endpoints in order of priority.
+func (r ResourceLink) GetUnsecureEndpoints() []Endpoint {
```

Comment:
Is it neccessary (required by some API) to have a separate functions for secure and unsecure? Couldn't it be unified into a single function with a (un)secure parameter?

Reply:
I refactored to use common function.